### PR TITLE
Support java8

### PR DESCRIPTION
**Pull request by GerardPaligot**
Spoon must support some things to be compatible with Java 8 :

- Annotations on types
- Annotations repeated on a element
- Default and static methods in interfaces
- Lambda expressions
- Executable reference expressions

So you can test this PR in your projects and let me feedback if you found bugs!

### CODE REVIEW

**Comment by pschichtel** [comment]
Any ETA on this?

**Comment by GerardPaligot** [comment]
What is ETA?

File: src/main/java/spoon/delegate/ExecutableDelegate.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package spoon.delegate;
+
+import spoon.reflect.code.CtBlock;
+import spoon.reflect.declaration.CtParameter;
+import spoon.reflect.reference.CtTypeReference;
+
+import java.util.List;
+import java.util.Set;
+
```
Comment (monperrus) [inline src/main/java/spoon/delegate/ExecutableDelegate.java]: Javadoc at the interface level (one sentence to describe the responsibility)

File: src/main/java/spoon/reflect/code/CtExecutableReferenceExpression.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package spoon.reflect.code;
+
+import spoon.reflect.reference.CtExecutableReference;
+
+public interface CtExecutableReferenceExpression<T, E extends CtExpression<?>> extends CtTargetedExpression<T, E> {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/CtExecutableReferenceExpression.java]: extends Targeted?

File: src/main/java/spoon/reflect/code/CtLambda.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import spoon.reflect.declaration.CtExecutable;
+
+/**
+ * This code element represents the creation of a lambda. A lambda
+ * can have two sorts of body : an simple expression or a block of
+ * statements. The usage of this concept in this class is:
+ *
+ * <ul>
+ *     <li>
+ *         If your lambda has an expression, getBody method will
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/CtLambda.java]: snippet?

File: src/main/java/spoon/reflect/code/CtLambda.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ *         return null and getExpression method will return a
+ *         CtExpression.
+ *     </li>
+ *     <li>
+ *         If your lambda has a block of statement, getExpression
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/CtLambda.java]: Snippet?

File: src/main/java/spoon/reflect/code/CtTypeAccess.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package spoon.reflect.code;
+
+import spoon.reflect.reference.CtTypeReference;
+
+/**
+ * This code element defines an access to a type.
+ *
```
Comment (monperrus) [inline src/main/java/spoon/reflect/code/CtTypeAccess.java]: snippet?

File: src/main/java/spoon/reflect/declaration/CtMethod.java [hunk 6]
```diff
@@ -20,6 +20,14 @@
 /**
  * This element defines a method declaration.
  */
-public interface CtMethod<T> extends CtExecutable<T>, CtTypedElement<T> {
+public interface CtMethod<T> extends CtExecutable<T>, CtTypeMember, CtGenericElement {
+	/**
+	 * Checks if the method is a default method.
```
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtMethod.java]: always in in interface, add a link to the official doc

File: src/main/java/spoon/reflect/reference/CtTypeAnnotableReference.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
+package spoon.reflect.reference;
+
+import spoon.reflect.declaration.CtAnnotation;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+
```
Comment (monperrus) [inline src/main/java/spoon/reflect/reference/CtTypeAnnotableReference.java]: javadoc?

Comment (monperrus) [inline src/main/java/spoon/reflect/reference/CtTypeAnnotableReference.java]: snippet?

File: src/test/java/spoon/test/TestUtils.java [hunk 8]
```diff
@@ -50,4 +53,17 @@ public static Factory build(Class<?>... classesToBuild) throws Exception {
 		return comp.getFactory();
 	}
 
+	public static boolean canBeBuild(File testDirectory, int complianceLevel) throws IOException {
```
Comment (monperrus) [inline src/test/java/spoon/test/TestUtils.java]: assertIsBuildable with throw new AssertError("good message') ?

File: .travis.yml [hunk 9]
```diff
@@ -1 +1,4 @@
 language: java
+
+jdk:
+  - oraclejdk8
```
Comment (monperrus) [inline .travis.yml]: add openjdk8

**Comment by pschichtel** [comment]
http://en.wikipedia.org/wiki/Estimated_time_of_arrival

**Comment by GerardPaligot** [comment]
@pschichtel: A snapshot should come quickly.

File: src/main/java/spoon/reflect/code/CtLambda.java (hunk 3 above)
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/code/CtLambda.java]: Done but not at this line.

File: src/main/java/spoon/reflect/reference/CtTypeAnnotableReference.java (hunk 7 above)
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/reference/CtTypeAnnotableReference.java]: Done but not at this line.

File: .travis.yml (hunk 9 above)
Comment (GerardPaligot) [inline .travis.yml]: Done but not at this line.

File: src/main/java/spoon/reflect/declaration/CtMethod.java (hunk 6 above)
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/declaration/CtMethod.java]: I don't understand what do you want. Can you explain it?

File: .travis.yml (hunk 9 above)
Comment (GerardPaligot) [inline .travis.yml]: openjdk isn't supported by travis : http://docs.travis-ci.com/user/languages/java/#Overview.

**Pull request merged**